libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -291,10 +292,28 @@ proc handleControl(g: GossipSub, peer: PubSubPeer, control: ControlMessage) =
         libp2p_pubsub_broadcast_prune.inc(labelValues = ["generic"])
 
     trace "sending control message", msg = shortLog(respControl), peer
-    g.send(
+    asyncSpawn g.send(
       peer,
       RPCMsg(control: some(respControl), messages: messages))
 
+proc lazyBroadcast(
+    g: GossipSub,
+    peers: seq[PubSubPeer],
+    msgIdSalted: MessageId,
+    msg: RPCMsg) {.async.} =
+  let
+    futsTable = g.cancellableBroadcast(peers, msg)
+    futs = toSeq(futsTable.values)
+  g.sendingFutures[msgIdSalted] = futsTable
```

Comment:
can't there be two inflight lazy broadcasts of the same message?

Reply:
That shouldn't happen, I may add an assert to be sure